generators/entity-server/templates/src/main/java/package/web/rest/EntityResource.java.ejs

```diff
@@ -111,7 +114,12 @@ public class <%= entityClass %>Resource {
         log.debug("REST request to save <%= entityClass %> : {}", <%= instanceName %>);
         if (<%= instanceName %>.getId() != null) {
             throw new BadRequestAlertException("A new <%= entityInstance %> cannot already have an ID", ENTITY_NAME, "idexists");
-        }<%- include('../../common/save_template', {viaService: viaService, returnDirectly: false}); -%>
+        }<%_ if (saveUserSnapshot) { %>
+        if (<%= instanceName %>.getUser() != null) {
+            // Save user in case it's new and only exists in gateway
+            userRepository.save(<%= instanceName %>.getUser());
```

Comment:
Shouldn't we verify that the login and id are also not null ?

Reply:
Also it fails if the entity uses DTOs as DTOs don't have a `User` attribute but `userId` and `userLogin` instead.

Reply:
Are you commenting on an old issue? You know that's not allowed, right? ;)

Sounds like a bug so probably best to open an issue. I don't think it's necessary w/o DTOs because the User is usually populated via dropdown.

Reply:
Yes, I'll open an issue later (no time right now...). It's just in case I forget...